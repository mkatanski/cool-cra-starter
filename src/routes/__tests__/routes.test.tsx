/* eslint-disable react/display-name */
import React, { Suspense } from "react";
import TestRenderer from "react-test-renderer";
import { wait, cleanup } from "@testing-library/react";

import Routes from "../routes";

jest.mock("routes/AuthenticatedRoute", () => ({
  AuthenticatedRoute: ({
    component: Component,
    ...rest
  }: {
    component: React.FunctionComponent;
  }) => (
    <div id="AuthenticatedRoute" {...rest}>
      <Component />
    </div>
  ),
}));

jest.mock("routes/UnauthenticatedRoute", () => ({
  UnauthenticatedRoute: ({
    component: Component,
    ...rest
  }: {
    component: React.FunctionComponent;
  }) => (
    <div id="UnauthenticatedRoute" {...rest}>
      <Component />
    </div>
  ),
}));

jest.mock(
  "pages/errors/NotFoundPage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="NotFoundPage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "pages/protected/HomePage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="HomePage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "pages/protected/EmptyPage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="EmptyPage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "pages/public/LoginPage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="LoginPage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "layouts/DefaultLayout",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="DefaultLayout" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "layouts/PublicLayout",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="PublicLayout" {...rest}>
      {children}
    </div>
  )
);

const waitForLazyImports = async (importsNumber: number) => {
  for (let i = 0; i < importsNumber; i++) {
    await wait();
  }
};

describe("routes", () => {
  beforeEach(cleanup);

  it("should render routes schema", async () => {
    const tree = TestRenderer.create(
      <Suspense fallback={"loading"}>
        <Routes />
      </Suspense>
    );

    await waitForLazyImports(2);

    expect(tree).toMatchSnapshot();
  });
});
