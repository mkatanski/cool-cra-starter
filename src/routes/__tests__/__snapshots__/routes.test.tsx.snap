// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should render routes schema 1`] = `
<div
  id="ReactRouter.Switch"
>
  <div
    id="PublicLayout"
    path={
      Array [
        "/login",
      ]
    }
  >
    <div
      allowAuthenticated={true}
      exact={true}
      id="UnauthenticatedRoute"
      path={
        Array [
          "/login",
        ]
      }
    >
      <div
        id="LoginPage"
      />
    </div>
  </div>
  <div
    id="DefaultLayout"
    path="/*"
  >
    <div
      exact={true}
      id="ReactRouter.Route"
      path="/"
    >
      <div
        id="ReactRouter.Redirect"
        to="/dashboard"
      />
    </div>
    <div
      exact={true}
      id="AuthenticatedRoute"
      path="/dashboard"
    >
      <div
        id="HomePage"
      />
    </div>
    <div
      exact={true}
      id="AuthenticatedRoute"
      path="/empty"
    >
      <div
        id="EmptyPage"
      />
    </div>
    <div
      exact={true}
      id="AuthenticatedRoute"
      path="/empty"
    >
      <div
        id="EmptyPage"
      />
    </div>
    <div
      exact={true}
      id="AuthenticatedRoute"
      path="/empty"
    >
      <div
        id="EmptyPage"
      />
    </div>
    <div
      id="AuthenticatedRoute"
    >
      <div
        id="NotFoundPage"
      />
    </div>
  </div>
  <div
    id="ReactRouter.Route"
  >
    <div
      id="NotFoundPage"
    />
  </div>
</div>
`;
